### DISCUSSION TITLE
Why is setting up a library necessary?

### DISCUSSION THREAD

**Discussion by HoutarouOreki** [spacedriveapp/spacedrive#2761] [General]
I don't know if I'm missing something. I want a file explorer. Why does it need me to setup some library before I can use it? I'm going to just enter random letters so it lets me skip past this screen.

![image](https://github.com/user-attachments/assets/08a90636-c398-4af6-ad93-3630380aef7e)

Edit: it's just useless anyway, doesn't open a single folder on my drive, then my whole PC lags out and this is what I see in task manager, even though spacedrive doesn't have any windows opened. Does anybody actually use this?
![image](https://github.com/user-attachments/assets/238f9f4d-7c81-4545-afb6-8078df89fbd0)

**Comment by iLynxcat** [comment]
Hi @HoutarouOreki, I understand that the purpose of the Library isn't explained well enough in the UI. We've just recently started work on a new onboarding flow that would better explain the purpose of a Library.

As of now, in the current Alpha release 0.4.2, adding folders to your Library will have Spacedrive create a quickly-searchable index of the files in those folders, optional tagging data, with thumbnails for media, even on removable drives. We're hard at work building out more of the planned advantages of synchronization of that index & instantaneous file-fetching on your other devices for the upcoming Beta 0.5.0 release.

Your issue brought to mind that it's be a good idea to enable people to use Spacedrive simply as a file browser for their local system if they don't want to use a Library. I'm going to note the idea down and we'll explore it as a future feature.

All our developers and thousands of people use Spacedrive regularly — it's really strange to hear you can't open any folders & the application lags your computer, that's not an issue I've seen someone have yet. Is there any chance you could upload a screen recording of this happening so we can see what's going wrong, either in reply here or via email if you'd prefer?
